luigi/scheduler.py

```diff
@@ -546,7 +546,8 @@ def add_task(self, worker, task_id, status=PENDING, runnable=True,
         * add additional workers/stakeholders
         * update priority when needed
         """
-        self.update(worker)
+        worker_id = kwargs['worker']
```

Comment:
Well, it will raise an error I suppose. It might be good for readability of code to place an assert there. Or in a comment.

Otherwise everything else looks fine.